src/EFCore.PG.NodaTime/Extensions/TypeExtensions.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using System.Collections.Generic;
+
+#nullable enable
+
+// ReSharper disable once CheckNamespace
+namespace Npgsql.EntityFrameworkCore.PostgreSQL
+{
+    internal static class TypeExtensions
+    {
+        internal static bool IsGenericList(this Type type)
+            => type.IsGenericType && type.GetGenericTypeDefinition() == typeof(List<>);
+
+        internal static bool IsArrayOrGenericList(this Type type)
+            => type.IsArray || type.IsGenericList();
+
+        internal static bool TryGetElementType(this Type type, out Type? elementType)
```

Comment:
It's better to return an element type as is or `null`.

Reply:
I'm not sure... This means that instead of:

```c#
if (!operand.Type.TryGetElementType(out var operandElementType))
```

we need to write:

```c#
if (!(operand.Type.GetElementType() is Type operandElementType))
```

which seems considerably more verbose/ugly. Another small advantage of TryGetElementType is that you can put an existing variable.

Both patterns seem to have advantages and disadvantages...